posthog/hogql/transforms/lazy_tables.py

```diff
@@ -203,13 +251,34 @@ def visit_select_query(self, node: ast.SelectQuery):
                 join_scope.from_table,
                 join_scope.to_table,
                 join_scope.fields_accessed,
+                join_constraint_overrides.get(to_table, {}),
                 self.context,
                 node,
             )
-            join_to_add = cast(ast.JoinExpr, clone_expr(join_to_add, clear_locations=True))
+
+            # Removes the resolved table from scope if we've already seen it
+            if join_to_add.alias is not None:
+                join_to_add_table_name = join_to_add.alias
+            elif (
+                join_to_add.table is not None
+                and isinstance(join_to_add.table, ast.Field)
+                and join_to_add.table.chain is not None
+                and len(join_to_add.table.chain) > 0
+            ):
+                join_to_add_table_name = str(join_to_add.table.chain[0])
+
+            if (
+                join_to_add_table_name is not None
+                and node.type is not None
+                and node.type.tables.get(join_to_add_table_name) is not None
+            ):
+                node.type.tables.pop(join_to_add_table_name)
```

Comment:
Could this cause problems in some case? Is there any time when a legit join with the same name could have been done to a table?

 E.g.
```sql
   select event, person.properties.email as real_email
     from events 
left join person as person 
       on (person.properties.email = events.properties.tr00_mail)
left join person as person2
       on (person2.properties.email = events.properties.fake_mail)
```

I haven't followed the code to verify if this would be an issue or not, but would this run?

Reply:
I believe in this example, this would be done on the alias and not the table itself, (e.g. `person` and `person2`). But I'll test this to verify